### ISSUE TITLE
Scoring Question: Group's Software

### ISSUE THREAD

**Issue by beerMT** [rabobank-cdc/DeTTECT#52]
When using the command line argument --software-group, there is a note that it does not influence the scores. If the TA group uses a software/tool, which in turn covers a Technique/Subtechnique, shouldn't the Software TTP be added into the score of the heat map?

I'm curious as to what the design decision was for that caveat to be made, I might be missing something with regard to my understanding on how to track a group's known TTPs. After running a few tests, it is clear that a Group's TTP JSON layer does not necessarily include the Software TTP layer. Since this is the case, it seems that we should be adding the two JSON layers together to better understand the operational practices of the TA.

Thanks for the clarification.

**Comment by rubinatorz** [comment]
hi @beerMT 

DeTT&CT uses the information on threat actors documented in ATT&CK (also available in their STIX/TAXII feed):
https://attack.mitre.org/groups/

MITRE ATT&CK has based their information on a bunch of open source intelligence reports. So the documented (sub) techniques are techniques that have been seen and reported on.

The DeTT&CT group mode will show exactly those techniques that are documented in ATT&CK. Those will get a score, so that you can see which techniques are used the most. Our thought was: when a group uses a piece of software, it also has capabilities to use techniques that are covered within that software but which they didn't use it before (at least, that's what we know based on the intel reports mentioned above). You can say: this is the potential of the TA, because they are using specific software. And that's what we want to show with the --software-group option. It doesn't influence the score, because based on what we now on the intel available in ATT&CK the TA didn't use the technique before, but it's capable of using it.

**Comment by beerMT** [comment]
Interesting interpretation. Please allow me to offer up some counters. This is only to help further the project and its TA coverage and should not be viewed as nitpicking. My understanding is the following:

Lets look at TA505 (G0092) or FIN6 (G0037). These TAs are commonly associated  with the Software they have developed in order to achieve their objectives. TA505 frequently leverages Trickbot and it would be remiss to not track the associated Trickbot techniques along with TA505's overall Techniques used. This example holds true for FIN6 (and many others) as they have been found to use Maze Software. No other group is attributed to the Maze Software. Therefore, it seems more appropriate to track both FIN6's known Techniques and Software used (which maps to techniques) to represent a more complete scope of what Techniques the TA actually will use against a target.

From the MITRE Software page, it seems fairly clear that if a Group is associated with a Software it is because there is publicly available information to support the use. 
![image](https://user-images.githubusercontent.com/6117308/125320539-0bf0da00-e2f9-11eb-8cf1-3129c355cb39.png)

My suggestion is that the Software used by a Group gets populated along with the Techniques used by a group, since their is evidence of the Group leveraging that Software in attacks. In much the same fashion that MITRE includes Techniques used by a group, it is not a guarantee that the TA will use these techniques every time, it is just a representation of what they have used in the past and provides insight into their behaviors.

Hopefully the feed back is useful. I have had to repurpose parts of attackcti and DeTT&CT in order to achieve this desired behavior of tracking "all" Techniques known to be used by a group, so this isn't an ask that I need it to be built. It is more of a discussion that I think there is more fidelity to be achieved in tracking what techniques a TA is actually leveraging against their targets.

Cheers! And as always - thank you for this wonderful contribution to the InfoSec community.

**Comment by marcusbakker** [comment]
Hi Brad,

Ruben and I have just discussed this and believe you have a valid point. We are thus also of the opinion that changing the default behaviour to include the software techniques in the counting will improve the TA functionality of DeTT&CT. 

We have put this improvement on our backlog and are considering providing the user with several options to influence the TA technique count behaviour.

Thank Brad for bringing this to our attention! 😄 

Regards,
Marcus

**Comment by beerMT** [comment]
I appreciate you all being open to feedback. Cheers!

**Comment by rubinatorz** [comment]
hi @beerMT! It took some time but we finally implemented your idea! We were working on the change for ATT&CK Campaigns and that was a good moment to also include your idea. Note from the release notes:

We added a new option to the Group mode: `--include-software`. Thanks to [beerMT](https://github.com/beerMT). He came up with the [idea](https://github.com/rabobank-cdc/DeTTECT/issues/52) to include software techniques in the scores of the heat map when a threat actor uses specific software. Until now we only had the option to show what software is used (`--software -group`), but that option did not influence the score.

Checkout the new version:
https://github.com/rabobank-cdc/DeTTECT/wiki/Changelog#version-180
